pkg/controller/metrics.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
```

Comment:
probably worth being in its own package with an OWNERS file owned by sig-instrumentation.

Reply:
Ack

Reply:
any update on this?

Reply:
@logicalhan I've just pushed some updates, apologies for the delay.

Reply:
> @logicalhan I've just pushed some updates, apologies for the delay.

We actually put these sorts of metrics into component-base, like so: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/component-base/metrics/prometheus

Reply:
@logicalhan: should I do component-base/metrics/controller?

Reply:
no it should be `/component-base/metrics/prometheus/controller`

Reply:
@logicalhan got it, i'll do that in my next work session. any feedback on the the metrics_test.go file in this PR?